Flaky test failure in circuit library TestNLocal.test_add_nlocal_1__5__3_

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues to confirm this idea does not exist. -->

### What is the expected enhancement?

From https://github.com/Qiskit/qiskit-terra/pull/5211: 
https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=20124&view=logs&j=cd283863-fb6a-5830-043c-9c2ff8e5eadb&t=2cfcf938-699c-5ca1-acca-22b32300fd1d&l=8487


```
test.python.circuit.library.test_nlocal.TestNLocal.test_add_nlocal_1__5__3_
---------------------------------------------------------------------------

Captured traceback:
      File "/home/vsts/work/1/s/test/python/circuit/library/test_nlocal.py", line 47, in assertCircuitEqual
    self.assertEqual(qc1, qc2)

      File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)

      File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/unittest/case.py", line 905, in _baseAssertEqual
    raise self.failureException(msg)

    AssertionError: <qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7f5858a787c0> != <qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x7f5858a4f1c0>


```

--- a/test/python/circuit/library/test_nlocal.py
+++ b/test/python/circuit/library/test_nlocal.py
@@ -96,14 +96,14 @@ class TestNLocal(QiskitTestCase):
         reference = QuantumCircuit(max(num_qubits))
 
         # construct the NLocal from the first circuit
-        first_circuit = random_circuit(num_qubits[0], depth)
+        first_circuit = random_circuit(num_qubits[0], depth, seed=4200)
         # TODO Terra bug: if this is to_gate it fails, since the QC adds an instruction not gate
         nlocal = NLocal(max(num_qubits), entanglement_blocks=first_circuit.to_instruction(), reps=1)
         reference.append(first_circuit, list(range(num_qubits[0])))
 
         # append the rest
         for num in num_qubits[1:]:
-            circuit = random_circuit(num, depth)
+            circuit = random_circuit(num, depth, seed=4200)
             nlocal.append(circuit, list(range(num)))
             reference.append(circuit, list(range(num)))
 
@@ -119,14 +119,14 @@ class TestNLocal(QiskitTestCase):
         reference = QuantumCircuit(max(num_qubits))
 
         # construct the NLocal from the first circuit
-        first_circuit = random_circuit(num_qubits[0], depth)
+        first_circuit = random_circuit(num_qubits[0], depth, seed=4220)
         # TODO Terra bug: if this is to_gate it fails, since the QC adds an instruction not gate
         nlocal = NLocal(max(num_qubits), entanglement_blocks=first_circuit.to_instruction(), reps=1)
         reference.append(first_circuit, list(range(num_qubits[0])))
 
         # append the rest
         for num in num_qubits[1:]:
-            circuit = random_circuit(num, depth)
+            circuit = random_circuit(num, depth, seed=4220)
             nlocal.add_layer(NLocal(num, entanglement_blocks=circuit, reps=1))
             reference.append(circuit, list(range(num)))
 
@@ -138,8 +138,8 @@ class TestNLocal(QiskitTestCase):
         num_qubits, depth = 2, 2
 
         # construct two circuits for adding
-        first_circuit = random_circuit(num_qubits, depth)
-        circuit = random_circuit(num_qubits, depth)
+        first_circuit = random_circuit(num_qubits, depth, seed=4242)
+        circuit = random_circuit(num_qubits, depth, seed=4242)
 
         # get a reference
         reference = first_circuit + circuit

